### PR TITLE
11. GET /api/reviews (queries)

### PR DESCRIPTION
**Pull request by AngelikaM-T**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -411,5 +411,96 @@ describe("GET - 200: /api/reviews/:review_id", () => {
...
+    });
+    test("GET - 400: Responds with an invalid order query if passed an invalid order query", () => {
+      return request(app)
+        .get("/api/reviews?order=increasing")
+        .expect(400)
+        .then(({ body }) => {
+          expect(body.msg).toBe("invalid order query");
+        });
+    });
+  });
```
Comment (callumu21) [inline __tests__/app.test.js:505]: It doesn't look like we have a test for valid order queries. You also need to check what should happen when the category query is valid (i.e. in the database) with no associated reviews, and what happens when the category query is invalid (i.e. not in the database at all).

File: __tests__/app.test.js [hunk 2]
```diff
@@ -411,5 +411,96 @@ describe("GET - 200: /api/reviews/:review_id", () => {
...
+            review_id: expect.any(Number),
+            review_img_url: expect.any(String),
+            title: expect.any(String),
+            votes: expect.any(Number),
+          });
+        });
+      });
+  });
+  describe("-error tests", () => {
+    test("GET - 404: responds with an error message, (Not found!), when passed an invalid endpoint", () => {
```
Comment (callumu21) [inline __tests__/app.test.js:481]: If you already tested this previously, you don't need this for each endpoint. It's an error that covers all endpoints, not just this one specifically

**Review by callumu21 — CHANGES_REQUESTED** [review]
A good start, but there are a few more test cases that need to be considered.

File: db/db.js [hunk 3]
```diff
@@ -21,19 +21,43 @@ exports.checkReviewExists = (review_id) => {
...
+      }
+    });
+};
+
+exports.checkCategoryExists = (category) => {
+  console.log(category)
+  if (category === undefined) {
+    return Promise.resolve("potato")
+  }
+  console.log("BYEEE")
```
Comment (pkaodev) [inline db/db.js:46]: 44 - change so it is Promise.resolve() (we arent doing anything with the return)
46 - remove console.log

File: models/game.models.js [hunk 4]
```diff
@@ -14,30 +18,54 @@ exports.selectCategories = () => {
...
+
+  if (!validSortQueries.includes(sort_by)) {
+    return Promise.reject({ status: 400, msg: "invalid sort query" });
+  }
+  if (!validOrderQueries.includes(order.toUpperCase())) {
+    return Promise.reject({ status: 400, msg: "invalid order query" });
+  }
+
+  return checkCategoryExists(category).then((testVariable) => {
+    console.log(testVariable)
```
Comment (pkaodev) [inline models/game.models.js:40]: remove testVariables

**Comment by pkaodev** [comment]
Just a couple of small changes the merge away!

**Pull request merged**